sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/parser/DataTypeParserSuite.scala

```diff
@@ -58,6 +58,7 @@ class DataTypeParserSuite extends SparkFunSuite {
   checkDataType("varchAr(20)", StringType)
   checkDataType("cHaR(27)", StringType)
   checkDataType("BINARY", BinaryType)
+  checkDataType("interval", CalendarIntervalType)
```

Comment:
nit: This change accepts an interval keyword in a schema string, so could you add tests in `StructTypeSuite`. too?
```
scala> org.apache.spark.sql.types.StructType.fromDDL("a interval")
res0: org.apache.spark.sql.types.StructType = StructType(StructField(a,CalendarIntervalType,true))
```